provisioning/service/src/EnrollmentGroupsClient.cs

```diff
@@ -122,7 +122,12 @@ public async Task<EnrollmentGroup> GetAsync(string enrollmentGroupId, Cancellati
         public Task DeleteAsync(string enrollmentGroupId, CancellationToken cancellationToken = default)
         {
             Argument.AssertNotNullOrWhiteSpace(enrollmentGroupId, nameof(enrollmentGroupId));
-            return DeleteAsync(new EnrollmentGroup(enrollmentGroupId, null), cancellationToken);
+
+            // Note that we are filling in an empty symmetric key attestation here regardless of what attestation
+            // this enrollment uses. This part of the enrollment object is not part of the service request that
+            // takes place within this overload, so there is no harm in this being the wrong attestation type
+            // and/or having the wrong keys.
+            return DeleteAsync(new EnrollmentGroup(enrollmentGroupId, new SymmetricKeyAttestation("", "")), cancellationToken);
```

Comment:
Oh, I think I see what you're doing here. We want to reuse the overload.

Reply:
In this case, it might be worth repeating the service API call. On one hand, we can create a "temp" enrollment (with proper comments that you have in here), on the other hand we could just repeat the service API call with the passed in values. I'd vote for the 2nd one since it is cleaner about what it is doing.

Reply:
You mean turn 
```csharp
public async Task DeleteAsync(EnrollmentGroup enrollment, CancellationToken cancellationToken = default);
```

into 

```csharp
public async Task DeleteAsync(string enrollmentGroupId, string ETag, CancellationToken cancellationToken = default);
```

Or did you mean something else?

To me, the DeleteAsync that takes the whole EnrollmentGroup is just a convenience method for users

Reply:
For what it's worth, I'm working to add the same ```ifUnchanged``` flag that we used in the IoT hub service client where it looks like:

```csharp
public virtual async Task DeleteAsync(string deviceId, CancellationToken cancellationToken = default)
{
    Argument.AssertNotNullOrWhiteSpace(deviceId, nameof(deviceId));
    await DeleteAsync(new Device(deviceId), default, cancellationToken).ConfigureAwait(false);
}

public virtual async Task DeleteAsync(Device device, bool onlyIfUnchanged = false, CancellationToken cancellationToken = default);

```

Reply:
No, I meant repeat the service API call in both, but your example gave me another idea.
Why not have both
```c#
public async Task DeleteAsync(EnrollmentGroup enrollment, CancellationToken cancellationToken = default);
```
and 
```c#
public async Task DeleteAsync(string enrollmentGroupId, string ETag, CancellationToken cancellationToken = default);
```

but instead of (2) calling (1), have (1) call (2).

That way, we're not creating new objects but simply using the properties off user-supplied object to pass on to the service API.

Reply:
That approach, while valid, is a departure from the pattern we exhibit elsewhere in the IoT hub service client. If the creating of a new object to call an overload is the main problem here, then I can go with your first idea and just not have either DeleteAsync overload call the other.

```csharp
public async Task DeleteAsync(string enrollmentId, CancellationToken cancellationToken = default)
{
    // arg check, cancellation token check, then make service request with the just the enrollmentId
}

public async Task DeleteAsync(EnrollmentGroup enrollment, CancellationToken cancellationToken = default)
{
    // arg check, cancellation token check, then make service request with the enrollmentId and the ETag (if present)
}
```

Reply:
Track 2 guidance does suggest we have a ```Delete``` method that takes the full item, FYI

<img width="582" alt="image" src="https://user-images.githubusercontent.com/28789848/195725075-53daa39f-0215-4c42-bfc3-f652cd8a3b9f.png">